core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/DealiasLCAFilter.scala

```diff
@@ -0,0 +1,83 @@
+package com.databricks.labs.remorph.parsers.snowflake.rules
+
+import com.databricks.labs.remorph.intermediate._
+
+class DealiasLCAFilter extends Rule[LogicalPlan] with IRHelpers {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transform { case project: Project =>
+    dealiasProject(project)
+  }
+
+  private def dealiasProject(project: Project): Project = {
+    project.input match {
+      case filter: Filter => dealiasProject(project, filter)
+      case _ => project
+    }
+  }
+
+  def dealiasProject(project: Project, filter: Filter): Project = {
+    val aliases = project.columns
+      .filter(col => col.isInstanceOf[Alias])
+      .map(col => col.asInstanceOf[Alias])
+      .filter(alias => alias.child.isInstanceOf[Id]) // TODO do we need to support more than that ?
```

Comment:
We should avoid isInstanceOf in favor of match

Reply:
I'd agree if we cared about more than 1 type, but in this case it would make the code much more complex without a clear benefit ?

Reply:
The benefit is not using asInstanceOf ;)

Reply:
how is that a clear benefit ?